webapp/src/js/directives/auth.js

```diff
@@ -8,71 +8,71 @@ angular.module('inboxDirectives').directive('mmAuth', function(
-      $q
-        .all(promises)
+    const updateVisibility = promises => {
+      return $q.all(promises)
         .then(function () {
           element.removeClass('hidden');
+          return true;
         })
         .catch(function (err) {
           if (err) {
             $log.error('Error checking authorization', err);
           }
           element.addClass('hidden');
+          return false;
         });
-    }
+    };
 
-    var checkAuthAny = function() {
-      element.addClass('hidden');
+    const staticChecks = () => {
+      const promises = [];
+      if (attributes.mmAuth) {
+        promises.push(Auth(attributes.mmAuth.split(',')));
+      }
 
-      var mmAuthAny = mmAuthAnyGetter(scope);
-      mmAuthAny = _.isArray(mmAuthAny) ? mmAuthAny : [mmAuthAny];
+      if (attributes.mmAuthOnline) {
+        promises.push(Auth.online($parse(attributes.mmAuthOnline)(scope)));
+      }
 
-      if (_.any(mmAuthAny, function(element) {
-        return element === true;
-      })) {
-        return element.removeClass('hidden');
+      if (!promises.length) {
+        return true;
       }
 
-      var permissionsGroups = mmAuthAny
-        .filter(function(element) {
-          return _.isArray(element) || _.isString(element);
-        })
-        .map(function(element) {
-          return (_.isArray(element) && _.flatten(element)) || [ element ];
-        });
+      return updateVisibility(promises);
+    };
 
-      if (!permissionsGroups.length) {
-        return;
-      }
+    const dynamicChecks = allowed => {
+      if (allowed && attributes.mmAuthAny) {
+        const mmAuthAnyGetter = $parse(attributes.mmAuthAny);
+        scope.$watch(mmAuthAnyGetter, () => {
+          let mmAuthAny = mmAuthAnyGetter(scope);
+          mmAuthAny = Array.isArray(mmAuthAny) ? mmAuthAny : [mmAuthAny];
 
-      Auth
-        .any(permissionsGroups)
-        .then(function() {
-          element.removeClass('hidden');
-        })
-        .catch(function (err) {
-          if (err) {
-            $log.error('Error checking authorization', err);
+          if (mmAuthAny.some(element => element === true)) {
+            element.removeClass('hidden');
+            return;
           }
-          element.addClass('hidden');
+
+          const permissionsGroups = mmAuthAny
+            .filter(element => Array.isArray(element) || _.isString(element))
+            .map(element => (Array.isArray(element) && _.flatten(element)) || [ element ]);
+
+          if (!permissionsGroups.length) {
+            element.addClass('hidden');
+            return;
+          }
+
+          return updateVisibility([ Auth.any(permissionsGroups) ]);
         });
+      }
     };
 
-    if (attributes.mmAuthAny) {
-      var mmAuthAnyGetter = $parse(attributes.mmAuthAny);
-      scope.$watch(mmAuthAnyGetter, checkAuthAny);
+    const result = staticChecks();
+    if (result === true) {
+      dynamicChecks(true);
+    } else {
+      result().then(dynamicChecks);
```

Comment:
This isn't as clean as I'd like but it makes the code synchronous when possible which makes testing cleaner.